test/run_test.py

```diff
@@ -677,11 +677,18 @@ def run_doctests(test_module, test_directory, options):
 
 
 def print_log_file(test: str, file_path: str, failed: bool) -> None:
+    num_lines = sum(1 for _ in open(file_path, 'rb'))
+    n = 300
     with open(file_path, "r") as f:
         print_to_stderr("")
         if failed:
-            print_to_stderr(f"PRINTING LOG FILE of {test} ({file_path})")
-            print_to_stderr(f.read())
+            print_to_stderr(f"Expand the folded group to see the beginning of the log file of {test}")
+            print_to_stderr(f"##[group]PRINTING BEGINNING OF LOG FILE of {test} ({file_path})")
```

Comment:
Nit: The group print is redundant when n < num_lines here, but it should be a very rare case I guess, i.e. failing right at the beginning